chart/kubeapps/templates/_helpers.tpl

```diff
@@ -175,6 +175,13 @@ Create name for the secrets related to an app repository
 apprepo-{{ .name }}-secrets
 {{- end -}}
 
+{{/*
+Create name for the secrets related to oauth2_proxy
+*/}}
+{{- define "kubeapps.oauth2_proxy-secret.name" -}}
+{{ template "kubeapps.fullname" . }}-oauth2
```

Comment:
General comment here: we are limited to 53-characters-names  (if I remember correctly). Since we rely on `kubeapps.fullname` we left a variable number of characters available. I mean, the basic things (i.e., deployments, etc) can fit in this 53-char limit but other k8s resource might not.

I think we should guarantee a safe number of characters, so that the users can be sure in advance their name won't silently break anything. 

(but this name is ok, it was just speaking my mind here :P)

Reply:
> so that the users can be sure in advance their name won't silently break anything.

I won't fail silently, the K8s API fails when trying to install stuff with invalid names:

```
▶ helm install this-is-a-very-long-time-to-ensure ./chart/kubeapps       
Error: ServiceAccount "this-is-a-very-long-time-to-ensure-kubeapps-internal-apprepository-job-postupgrade" is invalid: metadata.labels: Invalid value: "this-is-a-very-long-time-to-ensure-kubeapps-internal-apprepository-controller": must be no more than 63 characters
```

Reply:
Yes, the 'helm install` command would complain if we set a very long name. However,  there is a "grey side" here: it is not the install that is failing but the postinstall or other helm hooks. When I face this error again, I'll open an issue and discuss it there

Reply:
? I don't fully understand what you mean. If the `postinstall` fails is true that you need to uninstall the chart first before being able to try again.

Reply:
I didn't explain well, sorry! I'm not being able to reproduce it right now, but the general idea is, more or less, what this issue describes (https://github.com/kubernetes/kubernetes/issues/50850).
Anyway, don't worry, if I could replicate it again, I'll let you know.